pkg/cmd/server/origin/controller/image.go

```diff
@@ -0,0 +1,141 @@
+			Resource:  schema.GroupResource{Group: "build.openshift.io", Resource: "buildconfigs"},
+			Informer:  ctx.DeprecatedOpenshiftInformers.BuildConfigs().Informer(),
+			Store:     ctx.DeprecatedOpenshiftInformers.BuildConfigs().Indexer(),
+			TriggerFn: triggerbuildconfigs.NewBuildConfigTriggerIndexer,
+			Reactor:   &triggerbuildconfigs.BuildConfigReactor{Instantiator: bcInstantiator},
+		})
+	}
+	if !c.HasDeploymentsEnabled {
+		sources = append(sources, imagetriggercontroller.TriggerSource{
+			Resource:  schema.GroupResource{Group: "extensions", Resource: "deployments"},
+			Informer:  ctx.DeprecatedOpenshiftInformers.KubernetesInformers().Apps().V1beta1().Deployments().Informer(),
+			Store:     ctx.DeprecatedOpenshiftInformers.KubernetesInformers().Apps().V1beta1().Deployments().Informer().GetIndexer(),
+			TriggerFn: triggerannotations.NewAnnotationTriggerIndexer,
+			Reactor:   &triggerannotations.AnnotationReactor{Updater: updater, Copier: kapi.Scheme},
+		})
+	}
+	if !c.HasDaemonSetsEnabled {
+		sources = append(sources, imagetriggercontroller.TriggerSource{
+			Resource:  schema.GroupResource{Group: "extensions", Resource: "daemonsets"},
+			Informer:  ctx.DeprecatedOpenshiftInformers.KubernetesInformers().Extensions().V1beta1().DaemonSets().Informer(),
+			Store:     ctx.DeprecatedOpenshiftInformers.KubernetesInformers().Extensions().V1beta1().DaemonSets().Informer().GetIndexer(),
+			TriggerFn: triggerannotations.NewAnnotationTriggerIndexer,
+			Reactor:   &triggerannotations.AnnotationReactor{Updater: updater, Copier: kapi.Scheme},
+		})
+	}
+	if !c.HasStatefulSetsEnabled {
+		sources = append(sources, imagetriggercontroller.TriggerSource{
+			Resource:  schema.GroupResource{Group: "apps", Resource: "statefulsets"},
+			Informer:  ctx.DeprecatedOpenshiftInformers.KubernetesInformers().Apps().V1beta1().StatefulSets().Informer(),
+			Store:     ctx.DeprecatedOpenshiftInformers.KubernetesInformers().Apps().V1beta1().StatefulSets().Informer().GetIndexer(),
+			TriggerFn: triggerannotations.NewAnnotationTriggerIndexer,
+			Reactor:   &triggerannotations.AnnotationReactor{Updater: updater, Copier: kapi.Scheme},
+		})
+	}
+	if !c.HasCronJobsEnabled {
+		sources = append(sources, imagetriggercontroller.TriggerSource{
+			Resource:  schema.GroupResource{Group: "batch", Resource: "cronjobs"},
+			Informer:  ctx.DeprecatedOpenshiftInformers.KubernetesInformers().Batch().V2alpha1().CronJobs().Informer(),
+			Store:     ctx.DeprecatedOpenshiftInformers.KubernetesInformers().Batch().V2alpha1().CronJobs().Informer().GetIndexer(),
+			TriggerFn: triggerannotations.NewAnnotationTriggerIndexer,
+			Reactor:   &triggerannotations.AnnotationReactor{Updater: updater, Copier: kapi.Scheme},
+		})
+	}
+
+	go imagetriggercontroller.NewTriggerController(
+		broadcaster,
+		streamInformer,
+		lister,
+		sources...,
+	).Run(5, ctx.Stop)
+
+	return true, nil
+}
+
+// TODO: remove when generated informers exist
```

Comment:
They exists, but are not wired, yet (or not everywhere) iirc. This should be removed when we entirely switch to the generated informers. The question is do we want to make it happen for 3.6?